lib/govuk_publishing_components/presenters/machine_readable/faq_page_schema.rb

```diff
@@ -57,42 +57,22 @@ def question_and_answers(html)
 
         question = page.title
 
+        # rubocop:disable Style/IfInsideElse
         doc.xpath("html/body").children.each_with_object({}) do |element, q_and_as|
-          _q_and_as, question = recursive_question_and_answers(element, question, q_and_as)
-        end
-      end
-
-      def recursive_question_and_answers(element, question, q_and_as)
-        if is_a_question?(element)
-          question = element.text
-          q_and_as[question] = { anchor: element["id"] }
-        else
-          q_and_as = add_answer_to_question(q_and_as, element, question)
-          element.children.each do |child_element|
-            if child_element.element?
-              q_and_as, question = recursive_question_and_answers(child_element, question, q_and_as)
-            end
-          end
-        end
-
-        [q_and_as, question]
-      end
-
-      def add_answer_to_question(q_and_as, element, question)
-        if no_questions_in_subtree?(element)
-          if question_hash_is_unset?(q_and_as, question)
-            q_and_as[question] = { answer: element.to_s }
-          elsif answer_is_unset?(q_and_as, question)
-            q_and_as[question][:answer] = element.to_s
+          if question_element?(element)
+            question = element.text
+            q_and_as[question] = { anchor: element["id"] }
           else
-            q_and_as[question][:answer] << element.to_s
+            if question_hash_is_unset?(q_and_as, question)
```

Comment:
Sorry - I see this is only part of the revert but I wonder if we can switch this to elsif to avoid the need to add the linter enabling/disabling?

Reply:
Yeah, I looked at that too - will fix 👍

Reply:
Updated 👍